apps/aehyperchains/src/aehc_connector.erl

```diff
@@ -0,0 +1,98 @@
+-module(aehc_connector).
+
+-export([send_tx/1, get_block/1]).
+
+-export([tx/3, header/2, block/2]).
+
+-export([publish/2, subscribe/1]).
+
+-type connector() :: atom().
+
+-callback send_tx(Tx::aetx:tx()) -> binary().
+
+-callback get_block(Num::integer()) -> block().
+
+-export_type([tx/0, header/0, block/0]).
+
+-spec connector() -> connector().
+connector() ->
+    ok.
+
+%%%===================================================================
+%%%  Parent chain simplified proto
+%%%===================================================================
+
+-record(tx, {
+             sender :: binary(),
+             recipient :: binary(),
+             payload :: binary()
```

Comment:
Don't we want to specialize `payload` with the information about commitment we are interested in?
ie
```
-record(tx, {
             sender :: binary(),
             recipient :: binary(),
             payload :: {not_commitment, binary()} | commitment()
}).

-record(commitment, {
  child_sign :: binary(),
  child_block :: binary(),
  delegate :: binary()
}).
```

Reply:
The knowledge about commitment itself is preserved in Db related stuff, we decided that connector will be as simple as possible for flexibility purpouses